packages/services/api/src/modules/cdn/providers/cdn.provider.ts

```diff
@@ -1,17 +1,22 @@
 import { createHmac } from 'crypto';
 import type { Span } from '@sentry/types';
+import { crypto } from '@whatwg-node/fetch';
+import bcryptjs from 'bcryptjs';
 import { Inject, Injectable, Scope } from 'graphql-modules';
 import { HiveError } from '../../../shared/errors';
 import { sentry } from '../../../shared/sentry';
+import { AuthManager } from '../../auth/providers/auth-manager';
+import { TargetAccessScope } from '../../auth/providers/scopes';
 import { HttpClient } from '../../shared/providers/http-client';
 import { Logger } from '../../shared/providers/logger';
-import type { CDNConfig } from './tokens';
-import { CDN_CONFIG } from './tokens';
+import { type S3Config, S3_CONFIG } from '../../shared/providers/s3-config';
+import { Storage } from '../../shared/providers/storage';
+import { type CDNConfig, CDN_CONFIG } from './tokens';
 
 type CdnResourceType = 'schema' | 'supergraph' | 'metadata';
 
 @Injectable({
-  scope: Scope.Singleton,
+  scope: Scope.Operation,
```

Comment:
Why this was changed? this will recreate the provider on every GraphQL operation, is that intended?

Reply:
Yes it is because we inject the auth provider for doing access checks